src-plugins/mscAlgorithmPaint/mscAlgorithmPaintToolBox.cpp

```diff
@@ -608,36 +608,60 @@ void AlgorithmPaintToolBox::updateView()
 
     if (view)
     {
-        setCurrentView(qobject_cast<medAbstractImageView*>(view));
-
-        updateMouseInteraction();
-
-        medAbstractData* data = currentView->layerData(0);
-
-        if(data)
+        medAbstractImageView * v = qobject_cast<medAbstractImageView*>(view);
+        if (v)
         {
-            medImageMaskAnnotationData * existingMaskAnnData = dynamic_cast<medImageMaskAnnotationData *>(data);
-            if(!existingMaskAnnData)
+            // AlgorithmPaintToolBox does not work with meshes
+            for (unsigned int i=0 ; i<v->layersCount() ; ++i)
             {
-                setData( data );
+                medAbstractData *data = v->layerData(i);
+                if(!data || data->identifier().contains("vtkDataMesh"))
+                {
+                    qDebug()<<"AlgorithmPaintToolBox::updateView() do not use meshes in this toolbox.";
+                    setButtonsDisabled(true);
+                    isMeshData=true;
+                    return;
+                }
+                else
+                    isMeshData=false;
             }
+            setButtonsDisabled(false);
+            currentView=v;
+        }
+        updateMouseInteraction();
 
-            // Update cursor if the orientation change
-            connect(currentView, SIGNAL(orientationChanged()), this, SLOT(activateCustomedCursor()), Qt::UniqueConnection);
-
-            connect(currentView,SIGNAL(layerRemoved(unsigned int)),this,SLOT(clear()), Qt::UniqueConnection);
-
-            // Update cursor to new view
-            if ( this->m_strokeButton->isChecked() )
-            {
-                activateCustomedCursor(); // Add circular cursor for painting
-            }
-            else
+        if (!isMeshData)
+        {
+            if (currentView)
             {
-                deactivateCustomedCursor(); // Deactivate painting cursor
-            }
+                medAbstractData* data = currentView->layerData(0);
 
-            setButtonsDisabled(false);
+                if(data)
```

Comment:
since in this code we check every data in the `if (v)` condition, you do not need anymore to check it here :)